### PR TITLE
provide starter IST to newly provisoned accounts

### PR DESCRIPTION
**Pull request by dckc**
refs: #6067

## Description

 - add `provisionPool` contract; wire it up to the walletFactory as well as each PSM
 - add module account, `vbank/provision`, to hold IST as well as assets to sell on the/a PSM

TODO:
 - [ ] make initial amount governed (I prefer to do this in a follow-on PR)

### Security Considerations

In the contract: `makeBridgeProvisionTool` is factored out of the contract `start` function to keep the bridge handling stuff mostly separate from the asset-handling stuff.

Elsewhere, the use of bootstrap powers merits careful review.

### Documentation Considerations

We should provide end users with docs on the "pay 10 BLD to the community pool; get 0.25 IST for initial fees" model.

### Testing Considerations

Trading on the PSM is unit-tested.

The flow resulting in 0.25 IST going to the user was only tested manually. (Results are documented [below](https://github.com/Agoric/agoric-sdk/pull/6157#issuecomment-1241481631).)

### CODE REVIEW

**Comment by dckc** [comment]
## how to authorize a module account to receive funds?

```
packages/cosmic-swingset$ make SOLO_COINS=12345000000ibc/usdc1234 ACCT_ADDR=$pool fund-acct
Waiting for localhost:26657 to come live.... done!
../../golang/cosmos/build/agd \
  --home=t1/bootstrap --keyring-backend=test --from=bootstrap \
  tx bank send bootstrap agoric1megzytg65cyrgzs6fvzxgrcqvwwl7ugpt62346 12345000000ibc/usdc1234 \
  --gas=auto --gas-adjustment=1.2 --broadcast-mode=block --yes --chain-id=agoric
Error: rpc error: code = InvalidArgument desc = failed to execute message; message index: 0: agoric1megzytg65cyrgzs6fvzxgrcqvwwl7ugpt62346 is not allowed to receive funds: unauthorized: invalid request
```

### p.s. Answer: don't _disable_ it

**Comment by dckc** [comment]
## First end-to-end test worked!

 - Supply funds to the pool
    1. Benefactor sends USDC to the pool account
    2. provisionPool trades the USDC for IST
 - Pay 10 BLD for an Agoric smart wallet
    1. User requests a smart-wallet account
    2. provisionPool provides a smartWallet and sends 0.25 IST
    3. User sees balance

### Send USDC to the pool account

```
packages/cosmic-swingset$ make SOLO_COINS=1234000000ibc/usdc1234 ACCT_ADDR=$pool fund-acct
../../golang/cosmos/build/agd \
  --home=t1/bootstrap --keyring-backend=test --from=bootstrap \
  tx bank send bootstrap agoric1megzytg65cyrgzs6fvzxgrcqvwwl7ugpt62346 1234000000ibc/usdc1234 \
  --gas=auto --gas-adjustment=1.2 --broadcast-mode=block --yes --chain-id=agoric
gas estimate: 84493
code: 0
...
txhash: ED3DED583373BBDAA734E7122FE9BA0498C7F17B8CABB7747783317A8007E1E3
```

### provisionPool trades the USDC for IST

```
2022-09-09T03:57:52.490Z SwingSet: vat: v15: bank balance update { address: 'agoric1megzytg65cyrgzs6fvzxgrcqvwwl7ugpt62346', amount: '1234000000', denom: 'ibc/usdc1234' }
2022-09-09T03:57:52.502Z SwingSet: vat: v18: provisionPool balance update { brand: Object [Alleged: AUSD brand] {}, value: 1_234_000_000n }
2022-09-09T03:57:52.715Z block-manager: block 20 commit
2022-09-09T03:57:57.483Z block-manager: block 21 begin
2022-09-09T03:57:57.711Z block-manager: block 21 commit
2022-09-09T03:58:02.491Z block-manager: block 22 begin
2022-09-09T03:58:02.642Z SwingSet: vat: v15: bank balance update { address: 'agoric1megzytg65cyrgzs6fvzxgrcqvwwl7ugpt62346', amount: '1233876600', denom: 'uist' }
2022-09-09T03:58:02.650Z SwingSet: vat: v18: provisionPool balance update { brand: Object [Alleged: IST brand] {}, value: 1_233_876_600n }
```

### Provision a smart-wallet account

_NOTE: This will cost N BLD shortly._

```
packages/cosmic-swingset$ addr=$(cat t1/8000/ag-cosmos-helper-address)
packages/cosmic-swingset$ make ACCT_ADDR=$addr provision-acct
../../golang/cosmos/build/agd --chain-id=agoric \
  --home=t1/bootstrap --keyring-backend=test --from=bootstrap \
  tx swingset provision-one t1/8000 agoric1xezchds0qr3wddht3t6zyrc2cn78q4v3alu8qu SMART_WALLET \
  --gas=auto --gas-adjustment=1.2 --broadcast-mode=block --yes
gas estimate: 111214
code: 0
...
txhash: 807A2A9AC224E2B99900978AA96D25319B12388030DF74EB2F6759F80226F1F9
```

### provisionPool provides a smartWallet and sends 0.25 IST

**NOTE** special coordination around _essential_ purses, i.e. IST.

```
2022-09-09T03:58:22.549Z SwingSet: vat: v18: PLEASE_PROVISION agoric1xezchds0qr3wddht3t6zyrc2cn78q4v3alu8qu [ 'SMART_WALLET' ]
2022-09-09T03:58:22.550Z SwingSet: vat: v18: sendInitialPayment agoric1xezchds0qr3wddht3t6zyrc2cn78q4v3alu8qu
2022-09-09T03:58:22.649Z SwingSet: vat: v15: bank balance update { address: 'agoric1megzytg65cyrgzs6fvzxgrcqvwwl7ugpt62346', amount: '1233626600', denom: 'uist' }
2022-09-09T03:58:22.705Z SwingSet: vat: v18: provisionPool balance update { brand: Object [Alleged: IST brand] {}, value: 1_233_626_600n }
2022-09-09T03:58:22.720Z SwingSet: vat: v16: { ready: false, essentialBrands: { Fee: Object [Alleged: IST brand] {} }, brandPurses: Object [Alleged: mapStore] { snapshot: [Function: snapshotMap], has: [Function: has], get: [Function: get], init: [Function: init], set: [Function: set], delete: [Function: del], keys: [Function: keys], values: [Function: values], entries: [Function: entries], getSize: [Function: getSize], clear: [Function: clear] } }
2022-09-09T03:58:22.781Z block-manager: block 26 commit
2022-09-09T03:58:27.548Z block-manager: block 27 begin
2022-09-09T03:58:27.609Z SwingSet: vat: v16: agoric1xezchds0qr3wddht3t6zyrc2cn78q4v3alu8qu notified of new asset Object [Alleged: AUSD brand] {}
2022-09-09T03:58:27.692Z SwingSet: vat: v16: agoric1xezchds0qr3wddht3t6zyrc2cn78q4v3alu8qu notified of new asset Object [Alleged: BLD brand] {}
2022-09-09T03:58:27.760Z SwingSet: vat: v16: { ready: false, essentialBrands: { Fee: Object [Alleged: IST brand] {} }, brandPurses: Object [Alleged: mapStore] { snapshot: [Function: snapshotMap], has: [Function: has], get: [Function: get], init: [Function: init], set: [Function: set], delete: [Function: del], keys: [Function: keys], values: [Function: values], entries: [Function: entries], getSize: [Function: getSize], clear: [Function: clear] } }
2022-09-09T03:58:27.782Z SwingSet: vat: v16: agoric1xezchds0qr3wddht3t6zyrc2cn78q4v3alu8qu notified of new asset Object [Alleged: IST brand] {}
2022-09-09T03:58:27.796Z SwingSet: vat: v16: agoric1xezchds0qr3wddht3t6zyrc2cn78q4v3alu8qu receive { paymentE: Object [Alleged: IST payment] {}, paymentBrand: undefined }
2022-09-09T03:58:27.855Z block-manager: block 27 commit
2022-09-09T03:58:32.560Z block-manager: block 28 begin
2022-09-09T03:58:32.604Z SwingSet: vat: v16: { ready: false, essentialBrands: { Fee: Object [Alleged: IST brand] {} }, brandPurses: Object [Alleged: mapStore] { snapshot: [Function: snapshotMap], has: [Function: has], get: [Function: get], init: [Function: init], set: [Function: set], delete: [Function: del], keys: [Function: keys], values: [Function: values], entries: [Function: entries], getSize: [Function: getSize], clear: [Function: clear] } }
2022-09-09T03:58:32.622Z SwingSet: vat: v16: { brand: Object [Alleged: IST brand] {}, essential: true }
2022-09-09T03:58:32.686Z SwingSet: vat: v16: { ready: true, essentialBrands: { Fee: Object [Alleged: IST brand] {} }, brandPurses: Object [Alleged: mapStore] { snapshot: [Function: snapshotMap], has: [Function: has], get: [Function: get], init: [Function: init], set: [Function: set], delete: [Function: del], keys: [Function: keys], values: [Function: values], entries: [Function: entries], getSize: [Function: getSize], clear: [Function: clear] } }
2022-09-09T03:58:32.688Z SwingSet: vat: v16: { purse: Object [Alleged: Virtual Purse] {} }
2022-09-09T03:58:32.810Z SwingSet: vat: v15: bank balance update { address: 'agoric1xezchds0qr3wddht3t6zyrc2cn78q4v3alu8qu', amount: '250000', denom: 'uist' }
2022-09-09T03:58:32.822Z SwingSet: vat: v16: agoric1xezchds0qr3wddht3t6zyrc2cn78q4v3alu8qu received { brand: Object [Alleged: IST brand] {}, value: 250_000n }
2022-09-09T03:58:32.835Z SwingSet: vat: v18: provisionPool sent { brand: Object [Alleged: IST brand] {}, value: 250_000n } to agoric1xezchds0qr3wddht3t6zyrc2cn78q4v3alu8qu
2022-09-09T03:58:32.836Z SwingSet: vat: v18: provisioned agoric1xezchds0qr3wddht3t6zyrc2cn78q4v3alu8qu [ 'SMART_WALLET' ]
```

### Tada! money in my pocket

```
packages/cosmic-swingset$ agd query bank balances $addr
balances:
- amount: "250000"
  denom: uist
```

cc @JimLarson @dtribble @turadg @michaelfig @rowgraus @samsiegart @Tartuffo

File: packages/vats/src/core/startWalletFactory.js [hunk 1]
```diff
@@ -53,51 +74,45 @@ export const startWalletFactory = async ({
...
-      await E(creatorFacet).provideSmartWallet(
-        address,
-        bank,
-        myAddressNameAdmin,
-      );
-      console.info('provisioned', address, powerFlags);
-    },
+  const handler = await E(ppFacets.creatorFacet).makeHandler({
+    // @ts-expect-error XXX what is going on here???
+    bankManager,
```
Comment (dckc) [inline packages/vats/src/core/startWalletFactory.js]: any clues, @turadg ?

File: packages/vats/src/provisionPool.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+export const start = (zcf, privateArgs) => {
+  // TODO: make initial amount governed
+  const { perAccountInitialAmount } = zcf.getTerms();
+  fit(privateArgs, privateShape, 'provisionPool privateArgs');
+  const { poolBank } = privateArgs;
+  fit(perAccountInitialAmount, AmountShape, 'perAccountInitialAmount');
+  const { brand: poolBrand } = perAccountInitialAmount;
+  /** @type {ERef<Purse>} */
+  // @ts-expect-error vbank purse is close enough for our use.
+  const fundPurse = E(poolBank).getPurse(poolBrand);
```
Comment (dckc) [inline packages/vats/src/provisionPool.js:92]: did you run accross this, @turadg ? any clues?

File: packages/vats/src/core/startWalletFactory.js [hunk 3]
```diff
@@ -53,51 +74,45 @@ export const startWalletFactory = async ({
...
+      // POLA contract only needs to register for srcId='wallet'
+      // TODO consider a scoped attenuation of this bridge manager to just 'wallet'
+      bridgeManager,
+    },
+  );
+  walletFactoryStartResult.resolve(wfFacets);
 
-      const hubBefore = await tryLookup(address);
-      assert(!hubBefore, 'already provisioned');
+  // const addr = PROVISION_ACCT_TEST;
```
Comment (Chris-Hibbert) [inline packages/vats/src/core/startWalletFactory.js]: drop?

File: packages/vats/src/provisionPool.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  /**
+   * @param {{
+   *   bankManager: ERef<BankManager>,
+   *   namesByAddressAdmin: ERef<NameAdmin>,
+   *   walletFactory: ERef<import('@agoric/vats/src/core/startWalletFactory').WalletFactoryStartResult['creatorFacet']>
+   * }} io
+   */
+  const makeHandler = ({ bankManager, namesByAddressAdmin, walletFactory }) =>
+    Far('provisioningHandler', {
+      fromBridge: async (_srcID, obj) => {
```
Comment (Chris-Hibbert) [inline packages/vats/src/provisionPool.js:37]: `obj` doesn't say enough about what this param is for.

File: packages/vats/src/provisionPool.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+import { E, Far } from '@endo/far';
+// TODO: move to narrower package?
+import { observeIteration, observeNotifier } from '@agoric/notifier';
+import { AmountMath, AmountShape } from '@agoric/ertp';
+import { WalletName } from '@agoric/internal';
+import { makeMyAddressNameAdmin, PowerFlags } from './core/basic-behaviors.js';
+
+const { details: X, quote: q } = assert;
+
+const privateShape = harden({
```
Comment (Chris-Hibbert) [inline packages/vats/src/provisionPool.js]:
```suggestion
const privateArgShape = harden({
```

**Review by JimLarson — APPROVED** [review]
Golang changes look good!

File: packages/wallet/contract/test/devices.js [hunk 6]
```diff
@@ -2,6 +2,7 @@ import bundleCentralSupply from '@agoric/vats/bundles/bundle-centralSupply.js';
 import bundleMintHolder from '@agoric/vats/bundles/bundle-mintHolder.js';
 import bundleSingleWallet from '@agoric/vats/bundles/bundle-singleWallet.js';
 import bundleWalletFactory from '@agoric/vats/bundles/bundle-legacy-walletFactory.js';
+import bundleProvisionPool from '@agoric/vats/bundles/bundle-provisionPool.js';
```
Comment (dckc) [inline packages/wallet/contract/test/devices.js:5]: I'd like to stop maintaining `packages/wallet/contract`. @samsiegart @michaelfig do we need it any more?

@turadg bonus points if you beat me to it.

File: packages/vats/src/core/basic-behaviors.js [hunk 7]
```diff
@@ -186,13 +186,11 @@ export const makeBoard = async ({
 harden(makeBoard);
 
 /**
- * @param {NameAdmin} namesByAddressAdmin
  * @param {string} address
  */
-export const makeMyAddressNameAdmin = (namesByAddressAdmin, address) => {
+export const makeMyAddressNameAdmin = address => {
```
Comment (turadg) [inline packages/vats/src/core/basic-behaviors.js]: Add "Kit" suffix

File: packages/vats/src/core/startWalletFactory.js [hunk 8]
```diff
@@ -53,51 +74,45 @@ export const startWalletFactory = async ({
...
+  const addr = await E(bankManager).getModuleAccountAddress('vbank/provision');
+  console.log('provision pool', { addr });
+  const poolBank = E(bankManager).getBankForAddress(
+    addr || PROVISION_ACCT_TEST,
```
Comment (turadg) [inline packages/vats/src/core/startWalletFactory.js]: Debugging or intentional for master?

File: packages/vats/src/provisionPool.js (hunk 5 above)
Comment (turadg) [inline packages/vats/src/provisionPool.js]: privateArgsShape

File: packages/vats/src/provisionPool.js [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+ *
+ * @param {(address: string, depositFacet: ERef<DepositFacet>) => Promise<void>} sendInitialPayment
+ */
+const makeBridgeProvisionTool = sendInitialPayment => {
+  /**
+   * @param {{
+   *   bankManager: ERef<BankManager>,
+   *   namesByAddressAdmin: ERef<NameAdmin>,
+   *   walletFactory: ERef<import('@agoric/vats/src/core/startWalletFactory').WalletFactoryStartResult['creatorFacet']>
+   * }} io
```
Comment (turadg) [inline packages/vats/src/provisionPool.js:33]: Consider "powers"

File: packages/vats/src/provisionPool.js [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+        assert.equal(
+          obj.type,
+          'PLEASE_PROVISION',
+          X`Unrecognized request ${obj.type}`,
+        );
+        const { address, powerFlags } = obj;
+        console.info('PLEASE_PROVISION', address, powerFlags);
+
+        if (!powerFlags.includes(PowerFlags.SMART_WALLET)) {
+          return;
```
Comment (turadg) [inline packages/vats/src/provisionPool.js]: Why not error? I haven't looked closely to answer for myself, but would be good yo have an explanation comment here regardless.

**Review by turadg — COMMENTED** [review]
Cursory review from phone. Only blocker I see is the Kit name. I'll look into types when not afk but can merge as is imo.

File: packages/vats/src/provisionPool.js [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+  };
+
+  const creatorFacet = Far('Provisioning Pool creator', {
+    makeHandler: makeBridgeProvisionTool(sendInitialPayment),
+    initPSM: (brand, instance) => {
+      fit(brand, M.remotable('brand'));
+      fit(instance, M.remotable('instance'));
+      brandToPSM.init(brand, instance);
+    },
+  });
```
Comment (Chris-Hibbert) [inline packages/vats/src/provisionPool.js:168]: Using `makeHeapFarInstance()` moves the parameter checking into the interface declaration.  It's a few more lines, but not more complex. Your choice whether to switch or not.
```suggestion
  const creatorFacetI = M.makeInstance('provisionPool creator Facet', {
    makeHandler: M.call().returns(M.any()), // I didn't try to figure out this type.
    initPSM: M.call(BrandShape, InstanceHandleShape),
  });
  
  const creatorFacet = makeHeapFarInstance(
    'Provisioning Pool creator',
    creatorFacetI,
    {
    makeHandler: makeBridgeProvisionTool(sendInitialPayment),
    initPSM: (brand, instance) => {
      brandToPSM.init(brand, instance);
    },
  });
```

File: packages/vats/src/core/startWalletFactory.js [hunk 12]
```diff
@@ -15,35 +14,57 @@ const { details: X } = assert;
 // eslint-disable-next-line no-unused-vars
 const startFactoryInstance = (zoe, inst) => E(zoe).startInstance(inst);
 
+const StableUnit = BigInt(10 ** Stable.displayInfo.decimalPlaces);
+
+const PROVISION_ACCT_TEST = 'agoric1muveuvn0nka9g6mx9fs0uxls5pvnmd23376dwq';
```
Comment (Chris-Hibbert) [inline packages/vats/src/core/startWalletFactory.js]: Should this go away before merging?   If not, what is its role in production?

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]
This looks basically right.  I suggested some clean-ups.

File: packages/wallet/contract/test/devices.js (hunk 6 above)
Comment (turadg) [reply packages/wallet/contract/test/devices.js:5]: https://github.com/Agoric/agoric-sdk/pull/6134/commits/22ff6e8415a88d6d42a265913cde5ae482c2fa26

File: packages/vats/src/core/startWalletFactory.js (hunk 8 above)
Comment (dckc) [reply packages/vats/src/core/startWalletFactory.js]: Debugging. But might be needed for the sim chain. Hm.

File: packages/vats/src/provisionPool.js (hunk 10 above)
Comment (dckc) [reply packages/vats/src/provisionPool.js]: Not sure. I have a vague memory that I tried that and something bad happened. But I should look more.

File: packages/vats/src/core/startWalletFactory.js (hunk 1 above)
Comment (turadg) [reply packages/vats/src/core/startWalletFactory.js]: fix in 28b104b76bef2ef1683cdefee3e89183ecff39b0

File: packages/vats/src/provisionPool.js (hunk 2 above)
Comment (turadg) [reply packages/vats/src/provisionPool.js:92]: yes. VirtualPurse is what the bank returns and it's slightly different. For example, `withdraw` is sync on Purse and async on VirtualPurse. @michaelfig documented this, https://github.com/Agoric/agoric-sdk/blob/28b104b76bef2ef1683cdefee3e89183ecff39b0/packages/vats/test/test-vat-bank.js#L135-L142

Making ERTP methods accept remote payments seems worth doing soon. WDYT @michaelfig ?

File: packages/vats/src/provisionPool.js (hunk 4 above)
Comment (dckc) [reply packages/vats/src/provisionPool.js:37]: It's a payload coming over the bridge. Is `payload` significantly better?

This name is conventional. `ck fromBridge|grep .js|grep obj|wc -l` shows 15 occurrences.
Consistency seems reasonably valuable.

File: packages/vats/src/core/startWalletFactory.js (hunk 12 above)
Comment (dckc) [reply packages/vats/src/core/startWalletFactory.js]: it's gone

File: packages/vats/src/provisionPool.js (hunk 11 above)
Comment (dckc) [reply packages/vats/src/provisionPool.js:168]: tried that. `M.makeInstance()` doesn't seem to be defined. I tried `M.instance()` but got:

```
2022-09-11T20:45:28.396Z SwingSet: vat: v1: walletFactoryStartResult settled; remaining: [ 'client', 'provisionPoolStartResult' ]
2022-09-11T20:45:29.607Z SwingSet: ls: v18: Logging sent error stack (Error#1)
2022-09-11T20:45:29.608Z SwingSet: ls: v18: Error#1: unrecognize method guard { optional: [Function: optional], rest: [Function: rest], returns: [Function: returns] }
2022-09-11T20:45:29.608Z SwingSet: ls: v18: Error: unrecognize method guard (an object)
 at construct ()
 at Error (/home/connolly/projects/agoric-sdk/packages/SwingSet/src/supervisors/subprocess-xsnap/lockdown-subprocess-xsnap.js:6160)
 at makeError (/home/connolly/projects/agoric-sdk/packages/SwingSet/src/supervisors/subprocess-xsnap/lockdown-subprocess-xsnap.js:2717)
 at fail (/home/connolly/projects/agoric-sdk/packages/SwingSet/src/supervisors/subprocess-xsnap/lockdown-subprocess-xsnap.js:2844)
 at baseAssert (/home/connolly/projects/agoric-sdk/packages/SwingSet/src/supervisors/subprocess-xsnap/lockdown-subprocess-xsnap.js:2862)
 at interface (.../store/src/patterns/patternMatchers.js:1655)
 at start (.../vats/src/provisionPool.js:158)
 at startContract (.../zoe/src/contractFacet/zcfZygote.js:332)
```

File: packages/vats/src/provisionPool.js (hunk 10 above)
Comment (dckc) [reply packages/vats/src/provisionPool.js]: Now I remember: I was thinking we could register 2 handlers for `BRIDGE_ID.WALLET`, on for `SMART_WALLET` and one for `REMOTE_WALLET`. But now that I think about it, we can only register one handler.

**Comment by dckc** [comment]
oops... @JimLarson I didn't mean to re-request a review from you. wrong button

File: packages/vats/src/provisionPool.js (hunk 11 above)
Comment (Chris-Hibbert) [reply packages/vats/src/provisionPool.js:168]: Sorry, that should have been `M.interface()`.   Not crucial for this pass.

File: packages/vats/src/core/startWalletFactory.js [hunk 13]
```diff
@@ -15,35 +14,67 @@ const { details: X } = assert;
 // eslint-disable-next-line no-unused-vars
 const startFactoryInstance = (zoe, inst) => E(zoe).startInstance(inst);
 
+const StableUnit = BigInt(10 ** Stable.displayInfo.decimalPlaces);
```
Comment (turadg) [inline packages/vats/src/core/startWalletFactory.js:17]: wonder if units should go in with their Brand def (i.e. tokens.js in this case)

**Review by turadg — APPROVED** [review]

File: packages/vats/src/core/startWalletFactory.js (hunk 13 above)
Comment (dckc) [reply packages/vats/src/core/startWalletFactory.js:17]: Perhaps, if there are other static uses of it... but a quick search doesn't find any other occurrences of `10 ** Stable.displayInfo.decimalPlaces`.

File: packages/vats/src/provisionPool.js (hunk 4 above)
Comment (Chris-Hibbert) [reply packages/vats/src/provisionPool.js:37]: > Is payload significantly better?

It seems so to me.

File: packages/vats/src/core/startWalletFactory.js [hunk 14]
```diff
@@ -15,35 +14,67 @@ const { details: X } = assert;
...
+  const [bridgeManager, poolAddr] = await Promise.all([
     bridgeManagerP,
-    namesByAddressAdminP,
+    E(bankManager).getModuleAccountAddress('vbank/provision'),
   ]);
+  if (!bridgeManager) {
+    console.warn('startWalletFactory needs bridgeManager (not sim chain)');
+    return;
+  }
+  console.log('provision pool', { poolAddr });
```
Comment (Chris-Hibbert) [inline packages/vats/src/core/startWalletFactory.js:64]: Should this be dropped or turned into `trace()`?

**Review by Chris-Hibbert — APPROVED** [review]

**Comment by Chris-Hibbert** [comment]
I'm only removing my request for changes, not actually adding an explicit approval. Not intending to block submission. I didn't see another way to stop blocking than to approve, so that's the button I pushed.

**Comment by turadg** [comment]
@Chris-Hibbert I re-requested review from you to clear the Approve

File: packages/vats/src/core/startWalletFactory.js (hunk 14 above)
Comment (dckc) [reply packages/vats/src/core/startWalletFactory.js:64]: as much as any other logging we do, I suppose; clean-up is postponed to #5222 , IIUC.

**Review by turadg — APPROVED** [review]
Still LGTM. Please rebase to clear the `fixup!` commits.

**Pull request merged**